### ISSUE TITLE
SHAP Summary not returning feature value colors?

### ISSUE THREAD

**Issue by bbennett36** [shap/shap#50]
My SHAP summary plots are only displaying in red and it is not showing the feature value bar. I just upgraded shap to the newest package and also tried the '%config InlineBackend.figure_format = 'retina' because this added the feature value colors for me before.  
  
Do you know what would cause this?  
  
Here is how I am calling it - 
shap_values = lgb_model.predict(X_validate, pred_contrib=True)
shap.summary_plot(shap_values, X_validate.columns)  
  
X_validate is just a pandas dataframe from a train_test_split.  
  
SHAP version - 
Name: shap
Version: 0.12.0
Summary: A unified approach to explain the output of any machine learning model.
Home-page: http://github.com/slundberg/shap
Author: Scott Lundberg
Author-email: slund1@cs.washington.edu
License: MIT
Location: /home/bbennett/miniconda3/envs/ml/lib/python3.6/site-packages
Requires: numpy, matplotlib, scipy, tqdm, scikit-learn, pandas, iml

Machine OS - 
CentOS Linux release 7.4.1708 (Core)

**Comment by bbennett36** [comment]
I figured out the issue. LightGBM can handle 'category' types from pandas so the data frame that I was passing to the summary plot couldn't get the min/max.

It was failing here because it couldn't use the percentile function on the category data type - 
vmin = np.nanpercentile(features[:, i], 5)
vmax = np.nanpercentile(features[:, i], 95)

I don't know if you would want to add a check for this but here was my solution. I just had to convert everything to numeric again - 
X = X_validate.apply(pd.to_numeric).as_matrix()
